app/scripts/modules/core/naming/naming.service.ts

```diff
@@ -0,0 +1,90 @@
+import {module} from 'angular';
+
+export const NAMING_SERVICE = 'spinnaker.core.naming.service';
+
+export interface IComponentName {
+  application: string;
+  stack: string;
+  freeFormDetails: string;
+}
+
+export class NamingService {
+  private VERSION_PATTERN: RegExp = /(v\d{3})/;
+
+  public parseServerGroupName(serverGroupName: string): IComponentName {
+    const result = {
```

Comment:
The TS compiler does not complain about it, but it makes sense to include the type for consistency.